xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/ModelFactory.java

```diff
@@ -1081,14 +1088,20 @@ public JobLog toRestJobLog(Collection<LogEvent> logs, URI self)
     /**
      * Check if the given property should be exposed via REST.
      * @param restProperty the property to be read/written
+     * @param the document from which the property comes; for access checks
      * @return true if the property is considered accessible
      */
-    private boolean hasAccess(Property restProperty)
+    private boolean hasAccess(Property restProperty, Document doc)
     {
         if (PASSWORD_TYPE.equals(restProperty.getType())) {
             return authorizationManagerProvider.get().hasAccess(Right.ADMIN, xcontextProvider.get().getWikiReference());
         }
 
+        if (EMAIL_TYPE.equals(restProperty.getType())) {
+            return !mailConfigProvider.get().isObfuscateEmails() ||
+                authorizationManagerProvider.get().hasAccess(Right.EDIT, doc.getDocumentReference());
```

Comment:
missing some comment to explain it.

Does it mean the REST call fails in this case? It shouldn't. I think the idea is just to obfuscate the email.